content/tutorials/hello-world-node.md

```diff
@@ -35,320 +36,459 @@ Node.js installation varies depending on your operating system.
+$ cd fluvio-node-app
 ```
 
-Make nightly toolchain default:
+Run npm to create a **node project** and generate package.json file:
 
 ```bash
-$ rustup default nightly
-info: using existing install for 'nightly-x86_64-apple-darwin'
-info: default toolchain set to 'nightly-x86_64-apple-darwin'
-
-nightly-x86_64-apple-darwin unchanged - rustc 1.44.0-nightly (f509b26a7 2020-03-18)
+$ npm init -y
 ```
 
+### Updating the Project for TypeScript
 
-##### Install Node build tool
-
-The **nj-cli** is a tool that generates native Node interfaces for Rust APIs. Use  **cargo** to install nj-cli tool:
+The Fluvio Node.js client provides TypeScript definitions. Let's setup our demo application to use TypeScript.
 
 ```bash
-$ cargo install nj-cli
-Updating crates.io index
-...
-Installing /Users/user/.cargo/bin/nj-cli
-Installed package `nj-cli v0.1.2` (executable `nj-cli`)    
+# Install TypeScript locally to the project as a development dependency;
+$ npm i typescript -D
 ```
 
-**Congratulations**, your environment is ready for use!
+### Create a Source Directory
 
+Create a `src/` directory where we will write our TypeScript files.
 
-## Build a simple data streaming App
+```bash
+# Create a `src/` directory
+mkdir src
 
-This section provides a step-by-step on how to build a simple data streaming app using Node.js. If you'd like to download the demo app instead, skip ahead to [Download Fluvio Demo Apps](#download-node-demo-apps).
+# Create an index.ts file
+touch ./src/index.ts
+```
 
+### Add Fluvio Client library
 
-### Start a new Node project
+The client library exports TypeScript definitions and Node.js bindings to the native Fluvio client.
 
-Create a directory for **fluvio-node-app**:
+Use npm install to add <a href="https://www.npmjs.com/package/@fluvio/client" target="_blank">@fluvio/client</a> to the project:
 
 ```bash
-$ mkdir fluvio-node-app
-$ cd fluvio-node-app
+$ npm install @fluvio/client --save
 ```
+Check to make sure the `@fluvio/client` module is added to package.json. 
 
-Run npm to create a **node project** and generate package.json file:
-
-```bash
-$ npm init -y
-```
+### Update `package.json` Scripts and Main Entry File
 
 ```json
 {
   "name": "fluvio-node-app",
   "version": "1.0.0",
   "description": "",
-  "main": "index.js",
+  "main": "./dist/index.js",
+  "types": "./dist/index.d.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "npx tsc ./src/index.ts --outDir ./dist --declaration --declarationMap",
+    "producer": "node ./bin/index.js -- producer",
+    "consumer": "node ./bin/index.js -- consumer"
   },
   "keywords": [],
-  "author": "fluvio <user@fluvio.io> (fluvio.io)",
-  "license": "Apache 2.0"
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@types/node": "^14.11.2",
+    "typescript": "^4.0.3"
+  },
+  "dependencies": {
+    "@fluvio/client": "^5.0.0"
+  }
 }
 ```
 
+{{< idea >}}
```

Comment:
I'm not sure I'm a fan of putting this in an idea bubble

Reply:
commit 3a13337 now uses `{{< idea >}}` to illustrate code comments;